### PR TITLE
Safer startup.

### PR DESCRIPTION
**Pull request by BaronGreenback**
It looks like it would be possible for a plugin to crash the system by generating an exception in their IServerEntryPoint implementation.

All methods are executed by a Task.WhenAll. According to the docs, this only reports the last error generated, and if i'm reading the code right, would crash.

This PR introduces another exception catching layer between the two. This should enable failed plugins to log exceptions without a crashing occurring.

### CODE REVIEW

File: Emby.Server.Implementations/ApplicationHost.cs [hunk 1]
```diff
@@ -465,17 +465,36 @@ public async Task RunStartupTasksAsync()
...
+        {
+            IEnumerable<Task> value;
+            try
+            {
+                value = StartEntryPoints(entryPoints, isBeforeStartup);
+            }
+            catch (Exception ex)
+            {
+                Logger.LogError(ex, "Exception in implementing SafeEntyPoint interface.");
+                value = new List<Task>
```
Comment (crobibero) [inline Emby.Server.Implementations/ApplicationHost.cs:489]:
```suggestion
                value = new[]
```

No reason to create a list for this

**Review by Bond-009 — CHANGES_REQUESTED** [review]
> If any of the supplied tasks completes in a faulted state, the returned task will also complete in a Faulted state, where its exceptions will contain the aggregation of the set of unwrapped exceptions from each of the supplied tasks.

**Comment by BaronGreenback** [comment]
> > If any of the supplied tasks completes in a faulted state, the returned task will also complete in a Faulted state, where its exceptions will contain the aggregation of the set of unwrapped exceptions from each of the supplied tasks.

The alternative is to wrap Task.WhenAll in a try/catch exception.

**Comment by Bond-009** [comment]
> The alternative is to wrap Task.WhenAll in a try/catch exception.
I don't think it actually throws, does it? If I read it correctly the exceptions are stored in the returned `Task`
https://docs.microsoft.com/en-us/dotnet/api/system.threading.tasks.task.exception?view=netframework-4.7#System_Threading_Tasks_Task_Exception

**Comment by BaronGreenback** [comment]
ignore me. been reading **await Task.WhenAll**

**Pull request closed without merging**